### PR TITLE
Initial enviroment setup

### LINKED ISSUE #4: feature #1 Initial setup
# REQUIREMENTS

Initializing rails project and adding rubocop, sandi_meter, rubycritic and all project requirements

# DESIGN
N/A

# Acceptance Criteria
List all required acceptance criteria
- [x] Initial Rails project
- [x] Rubocop
- [x] SandiMeter
- [x] RubyCritic
- [ ] PostgreSQL
- [x] Sass as css framework
- [x] Rspec as test framework

### PR DESCRIPTION
**Pull request by SpidySamurai**
# Description
We refactorized the setup project to generate atomic commits, in the first PR we send all changes in one commit,

We add the following tools:

- Rubocop
- Sandi_Meter
- Rubycritic
- Rspec
- Bootstrap forms
 
We also add the gitignore to the project.

We run the bundle install to add gems, and prove that test tools run correctly.
 
 # Testing
N/A
 
## Screenshots 
N/A

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+gem "rubocop", "~> 1.12"
+
+gem "sandi_meter", "~> 1.2"
+
+gem "rubycritic", "~> 4.6"
+
+gem "rspec", "~> 3.10"
```
Comment (anxhe) [inline Gemfile:64]: As a suggestion I would move rspec to test and development  group, as recomended me @marianoalarcon  👍

File: config/database.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# SQLite. Versions 3.8.0 and up are supported.
+#   gem install sqlite3
+#
+#   Ensure the SQLite 3 gem is defined in your Gemfile
+#   gem 'sqlite3'
+#
+default: &default
+  adapter: sqlite3
```
Comment (anxhe) [inline config/database.yml:8]: I think it is a good idea not to complicate with  postgrees, until the deployment to heroku

**Review by anxhe — COMMENTED** [review]
Good job friend only remember to import booststrap_form

**Review by anxhe — DISMISSED** [review]

File: Gemfile (hunk 1 above)
Comment (anxhe) [reply Gemfile:64]: Also , we are using rspec-rails gem instead

**Review by anxhe — COMMENTED** [review]
Uy,  I noticed you have conflicts. Let me know and I can check again when you solve them.

File: Gemfile (hunk 1 above)
Comment (Arkamis) [reply Gemfile:64]: roger, I think we can change this later when we create tests. Will check with  @SpidySamurai. Thanks you!

Comment (RobertVel666) [reply Gemfile:64]: There are some code style issues and you do seem to be missing some sass-rails and small-badge gems, but overall PR description checks in.

File: .gitignore [hunk 3]
```diff
@@ -34,3 +43,4 @@
 
 tmp
```
Comment (RobertVel666) [inline .gitignore]: Se repite tmp.

File: package.json [hunk 4]
```diff
@@ -2,10 +2,16 @@
...
   "dependencies": {
+    "@rails/actioncable": "^6.0.0",
+    "@rails/activestorage": "^6.0.0",
     "@rails/ujs": "^6.0.0",
+    "@rails/webpacker": "5.2.1",
+    "bootstrap": "^5.0.0-beta3",
     "turbolinks": "^5.2.0",
-    "@rails/activestorage": "^6.0.0",
-    "@rails/actioncable": "^6.0.0"
+    "yarn": "^1.22.10"
```
Comment (RobertVel666) [inline package.json:11]: is this necessary?

**Review by RobertVel666 — DISMISSED** [review]
Overall the little issues, the general config seems to be fine.

p.s.: did you guys run Rubocop on all the files and then change it back?

File: Gemfile (hunk 1 above)
Comment (SpidySamurai) [reply Gemfile:64]: I'll move it! Thank you, i noticed that robert we were missing 2 gems, that's true.

File: config/database.yml (hunk 2 above)
Comment (SpidySamurai) [reply config/database.yml:8]: Yes! Well change later when we get the config from postgres

File: package.json (hunk 4 above)
Comment (SpidySamurai) [reply package.json:11]: I'm not sure, it was generated automatically:c

**Comment by SpidySamurai** [comment]
I'm going to update the little issues before merging, thank you all!

**Review by Arkamis — APPROVED** [review]
LGTM!, We have some files and configurations that rails generated by default, however we will change them as we create functionalities, or as we use them.

File: package.json (hunk 4 above)
Comment (RobertVel666) [reply package.json:11]: I talked to Anxhe and she said it isn't necessary.

File: Gemfile (hunk 1 above)
Comment (RobertVel666) [reply Gemfile:64]: I would recommend moving the gems to their corresponding group.

**Review by RobertVel666 — APPROVED** [review]
It all seems fine, but if your using Yarn do you need a Package-lock.json?

**Pull request merged**